Update to use authselect for RHEL8

billglick opened the pull request:
Replace deprecated authconfig with authselect for EL8 (default)
Default to using authselect, with fallback to authconfig for EL7
Remove authconfig pkg for non-EL7
Update workflows and documentation
Update enable_mkhomedir parameter name

andylytical committed:
Update to use authselect for RHEL8

Replace deprecated authconfig with authselect for EL8 (default)
Default to using authselect, with fallback to authconfig for EL7; reenable oddjob for mkhomedir
Remove authconfig pkg for non-EL7
Update workflows and documentation
Update enable_mkhomedir parameter name
Add profile_system_auth::config::authselect_profile parameter
Set auth selection before setting features
Move RedHat specific data to ./data/os/RedHat.yaml

billglick commented:
This is being tested on the following hosts:
- asd-pup-client01.ncsa.illinois.edu (CentOS 7.9)
- asd-test-rhel8a.internal.ncsa.edu
- glick-client01.ncsa.illinois.edu
- glick-pup.ncsa.illinois.edu

andylytical commented:
Before merge, I think we need to resolve what should happen when both profile_system_auth and profile_pam_access are included on the same node. Both modules attempt to manage authselect / authconfig settings.

billglick commented:
> Before merge, I think we need to resolve what should happen when both profile_system_auth and profile_pam_access are included on the same node. Both modules attempt to manage authselect / authconfig settings.

We decided to remove the `authselect` profile selection from https://github.com/ncsa/puppet-pam_access/tree/ASDPLNG-49/rhel8 and instead do that only in profile_system_auth. So, I have added a `profile_system_auth::config::authselect_profile` parameter and updated logic in `manifests/config.pp` for `authselect` to select that profile.

andylytical commented:
@glick is this ready for review?

billglick commented:
@andylytical Yes, this is ready for review.

andylytical commented on data/common.yaml:
```diff
@@ -24,10 +25,14 @@ profile_system_auth::config::oddjobd_mkhomedir_conf: |
     </service>
   </oddjobconfig>
-profile_system_auth::config::required_pkgs:
+profile_system_auth::config::removed_pkgs:
+  - "fprintd-pam"
   - "authconfig"
+profile_system_auth::config::required_pkgs:
+  - "authselect"
   - "oddjob"
   - "oddjob-mkhomedir"
+profile_system_auth::config::use_authconfig: false
```
Minor, non-blocking suggestion: I think it makes sense that these lines be in data/os/RedHat.yaml (since the equivalent lines for EL7 are also under the data/os directory in hiera.

andylytical reviewed (approved):
LGTM.
One minor suggestion in data/common.yaml (see inline comment)

billglick commented on data/os/RedHat.yaml:
```diff
@@ -0,0 +1,14 @@
+---
+profile_system_auth::config::removed_pkgs:
+  - "fprintd-pam"
+  - "authconfig"
+profile_system_auth::config::required_pkgs:
+  - "authselect"
+  - "oddjob"
+  - "oddjob-mkhomedir"
+profile_system_auth::config::use_authconfig: false
+profile_system_auth::kerberos::required_pkgs:
+  - "krb5-libs"
+  - "krb5-workstation"
+profile_system_auth::ldap::required_pkgs:
+  - "openldap-clients"
```
I have now moved all of the RedHat specific parameters out of the common.yaml file and into the ./os/RedHat.yaml file.
